### PR TITLE
Fix errors bug on admin payment method form

### PR DESCRIPTION
**Pull request by vinay-mittal**
Name validation is in PaymentMethod class so not required in any of its inherited classes otherwise there will be double validation fail error.

### CODE REVIEW

File: backend/app/views/spree/admin/payment_methods/edit.html.erb [hunk 1]
```diff
@@ -7,8 +7,8 @@
 <%= form_for @payment_method, :url => admin_payment_method_path(@payment_method) do |f| %>
   <fieldset>
     <%= render :partial => 'form', :locals => { :f => f } %>
-    <div data-hook="buttons" class="actions">
-      <%= button Spree.t('actions.update'), 'save' %>
+    <div data-hook="admin_shipping_method_edit_form_buttons">
+      <%= render :partial => 'spree/admin/shared/edit_resource_links' %>
```
Comment (tanmay3011) [inline backend/app/views/spree/admin/payment_methods/edit.html.erb]: @vinayvinsol Use new ruby syntax. Avoid using '=>'.

File: core/app/models/spree/gateway.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ class Gateway < PaymentMethod
 
     delegate :authorize, :purchase, :capture, :void, :credit, to: :provider
 
-    validates :name, :type, presence: true
+    validates :type, presence: true
```
Comment (tanmay3011) [inline core/app/models/spree/gateway.rb]: @vinayvinsol Why did you remove presence validation on name?

Comment (jitendra) [inline core/app/models/spree/gateway.rb:7]: why we removed name from presence validation

Comment (vinay-mittal) [inline core/app/models/spree/gateway.rb:7]: Because its already present in the PaymentMethod class from which this one is inherited and then it will run twice

**Comment by jitendra** [comment]
Good. Please create direct PR
